src/test/scala/com/cognite/sdk/scala/common/DataPointsResourceBehaviors.scala

```diff
@@ -7,29 +7,61 @@ import org.scalatest.{FlatSpec, Matchers}
 
 trait DataPointsResourceBehaviors[I] extends Matchers { this: FlatSpec =>
   private val startTime = System.currentTimeMillis()
-  private val endTime = startTime + 20*1000
-  private val testDataPoints = (startTime to endTime by 1000).map(t =>
-    DataPoint(Instant.ofEpochMilli(t), math.random))
+  private val endTime = startTime + 20 * 1000
+  private val testDataPoints =
+    (startTime to endTime by 1000).map(t => DataPoint(Instant.ofEpochMilli(t), math.random))
+  private val granularity = "1s"
+  private val aggregates = Seq("average", "stepInterpolation")
 
   def withTimeSeriesId(testCode: I => Any): Unit
 
-  def dataPointsResource(dataPoints: DataPointsResource[Id, I]): Unit = {
-    it should "be possible to insert and delete numerical data points" in withTimeSeriesId { timeSeriesId =>
-      dataPoints.insertById(timeSeriesId, testDataPoints)
+  def dataPointsResource(dataPoints: DataPointsResource[Id, I]): Unit =
+    it should "be possible to insert and delete numerical data points" in withTimeSeriesId {
+      timeSeriesId =>
+        dataPoints.insertById(timeSeriesId, testDataPoints)
 
-      Thread.sleep(3000)
-      val points = dataPoints.queryById(timeSeriesId, startTime, endTime + 1)
-      points should have size testDataPoints.size.toLong
+        Thread.sleep(3000)
+        val points = dataPoints.queryById(timeSeriesId, startTime, endTime + 1)
+        points should have size testDataPoints.size.toLong
 
-      val latest = dataPoints.getLatestDataPointById(timeSeriesId)
-      latest.isDefined should be (true)
-      val latestPoint = latest.get
-      testDataPoints.toList should contain (latestPoint)
+        val latest = dataPoints.getLatestDataPointById(timeSeriesId)
+        latest.isDefined should be(true)
+        val latestPoint = latest.get
+        testDataPoints.toList should contain(latestPoint)
 
-      dataPoints.deleteRangeById(timeSeriesId, startTime, endTime + 1)
-      Thread.sleep(10000)
-      val pointsAfterDelete = dataPoints.queryById(timeSeriesId, startTime, endTime + 1)
-      pointsAfterDelete should have size 0
+        val aggregatePointMap =
+          dataPoints.queryAggregatesById(
+            timeSeriesId,
+            startTime,
+            endTime,
+            granularity,
+            aggregates
+          )
+        aggregatePointMap("average") should have size testDataPoints.size.toLong
+        aggregatePointMap("stepInterpolation") should have size testDataPoints.size.toLong - 1
+        aggregatePointMap.keys should not contain ("sum")
```

Comment:
Use `should contain theSameElementsAs` for better error messages: http://www.scalatest.org/user_guide/using_matchers#workingWithAggregations

We should also check the timestamps of the first and last data points for each aggregation (they should be ordered).